ogc-rs/src/gx.rs

```diff
@@ -0,0 +1,271 @@
+    pub fn clear_vtx_desc() {
+        unsafe { ogc_sys::GX_ClearVtxDesc() }
+    }
+
+    /// Sets the type of a single attribute (attr) in the current vertex descriptor.
+    /// See [GX_SetVtxDesc](https://libogc.devkitpro.org/gx_8h.html#af41b45011ae731ae5697b26b2bf97e2f) for more.
+    pub fn set_vtx_desc(attr: u8, v_type: u8) {
+        unsafe { ogc_sys::GX_SetVtxDesc(attr, v_type) }
+    }
+
+    /// Used to load a 3x4 modelview matrix mt into matrix memory at location pnidx.
+    /// See [GX_LoadPosMtxImm](https://libogc.devkitpro.org/gx_8h.html#a90349e713128a1fa4fd6048dcab7b5e7) for more.
+    pub fn load_pos_mtx_imm(mut mt: Mtx34, pnidx: u32) {
+        unsafe { extra_sys::GX_LoadPosMtxImm(&mut mt, pnidx) }
+    }
+
+    /// Sends a DrawDone command to the GP and stalls until its subsequent execution.
+    /// See [GX_DrawDone](https://libogc.devkitpro.org/gx_8h.html#a00f07b60ae2124fe027a82d7d9ae64b0) for more.
+    pub fn draw_done() {
+        unsafe { ogc_sys::GX_DrawDone() }
+    }
+
+    /// Sets the Z-buffer compare mode.
+    /// See [GX_SetZMode](https://libogc.devkitpro.org/gx_8h.html#a2af0d050f56ef45dd25d0db18909fa00) for more.
+    pub fn set_z_mode(enable: u8, func: u8, update_enable: u8) {
+        unsafe { ogc_sys::GX_SetZMode(enable, func, update_enable) }
+    }
+
+    /// Determines how the source image, generated by the graphics processor, is blended with the Embedded Frame Buffer (EFB).
+    /// See [GX_SetBlendMode](https://libogc.devkitpro.org/gx_8h.html#a1d9c43b161f3c5a30b9fd8ea182c8eb6) for more.
+    pub fn set_blend_mode(b_type: u8, src_fact: u8, dst_fact: u8, op: u8) {
+        unsafe { ogc_sys::GX_SetBlendMode(b_type, src_fact, dst_fact, op) }
+    }
+
+    /// Enables or disables alpha-buffer updates of the Embedded Frame Buffer (EFB).
+    /// See [GX_SetAlphaUpdate](https://libogc.devkitpro.org/gx_8h.html#ac238051bda896c8bb11802184882a2a0) for more.
+    pub fn set_alpha_update(enable: u8) {
+        unsafe { ogc_sys::GX_SetAlphaUpdate(enable) }
+    }
+
+    /// Enables or disables color-buffer updates when rendering into the Embedded Frame Buffer (EFB).
+    /// See [GX_SetColorUpdate](https://libogc.devkitpro.org/gx_8h.html#a3978e3b08198e52d7cea411e90ece3e5) for more.
+    pub fn set_color_update(enable: u8) {
+        unsafe { ogc_sys::GX_SetColorUpdate(enable) }
+    }
+
+    /// Sets the array base pointer and stride for a single attribute.
+    /// See [GX_SetArray](https://libogc.devkitpro.org/gx_8h.html#a5164fc6aa2a678d792af80d94bfa1ec2) for more.
+    pub fn set_array(attr: u32, ptr: *mut c_void, stride: u8) {
+        unsafe { ogc_sys::GX_SetArray(attr, ptr, stride) }
+    }
+
+    /// Begins drawing of a graphics primitive.
+    /// See [GX_Begin](https://libogc.devkitpro.org/gx_8h.html#ac1e1239130a33d9fae1352aee8d2cab9) for more.
+    pub fn begin(primitive: u8, vtxfmt: u8, vtxcnt: u16) {
+        unsafe { ogc_sys::GX_Begin(primitive, vtxfmt, vtxcnt) }
+    }
+
+    pub fn flush() {
```

Comment:
These could be implemented on the ``Drop`` of the whole structure to make sure they are ran when everything is freed.

Reply:
Call `flush()` when `Gx` goes out of scope?

Reply:
After looking into the GX system a bit more this requested change can be ignored unless GX has a function that should always be ran at the end.